guildwars2/characters.py

```diff
@@ -256,6 +257,229 @@ def handle_duplicates(upgrades):
+                            # Rune counter
+                            if upgrade in runes:
+                                runes[upgrade] += 1
+                            else:
+                                runes[upgrade] = 1
+            # Infusions
+            if "infusions" in piece:
+                if piece["slot"] not in ignore_list:
+                    infusions = piece["infusions"]
+                    for infusion in infusions:
+                        item_infusion = await self.fetch_item(infusion)
+                        if "infix_upgrade" in item_infusion["details"]:
+                            attributes = item_infusion["details"][
+                                "infix_upgrade"]["attributes"]
+                            for attribute in attributes:
+                                attr_dict[attribute["attribute"]] += attribute[
+                                    "modifier"]
+        for rune, runecount in runes.items():
+            rune_item = await self.fetch_item(rune)
+            bonuses = rune_item["details"]["bonuses"]
+            count = 0
+            for bonus in bonuses:
+                if count < runecount:
+                    pattern_single = re.compile("^\+\d{1,} ")
+                    pattern_all_stats = re.compile(".* [s,S]tats$")
+                    pattern_percentage = re.compile("^\+\d{1,}% ")
+                    # Regex deciding if it's a stat
+                    if pattern_all_stats.match(bonus):
+                        modifier = re.sub(' .*$', '', bonus)
+                        modifier = re.sub('\+', '', modifier)
+                        attr_dict["Power"] += int(modifier)
+                        attr_dict["Vitality"] += int(modifier)
+                        attr_dict["Toughness"] += int(modifier)
+                        attr_dict["Precision"] += int(modifier)
+                        attr_dict["Ferocity"] += int(modifier)
+                        attr_dict["Healing"] += int(modifier)
+                        attr_dict["ConditionDamage"] += int(modifier)
+                    elif pattern_single.match(bonus):
+                        # Regex deciding the attribute name + modifier
+                        modifier = re.sub(' .*$', '', bonus)
+                        modifier = re.sub('\+', '', modifier)
+                        attribute_name = re.sub('^.* ', '', bonus)
+                        if attribute_name in attr_dict:
+                            attr_dict[attribute_name] += int(modifier)
+                    elif pattern_percentage.match(bonus):
+                        modifier = re.sub(' .*$', '', bonus)
+                        modifier = re.sub('\+', '', modifier)
+                        attribute_name = re.sub(' Duration', 'Duration', bonus)
+                        attribute_name = re.sub('^.* ', '', attribute_name)
+                        if attribute_name in attr_dict:
+                            attr_dict[attribute_name] += int(modifier)
```

Comment:
Raises a ValueError since modifier can be a value with a percentage - eg `"5%"`

Reply:
Removed % with re.sub